### ISSUE TITLE
Bug: AWS plugin config: BSL Region not required when `s3ForcePathStyle` is `false`

### ISSUE THREAD

**Issue by kaovilai** [openshift/oadp-operator#424]
### Describe bug

When AWS plugin has `s3ForcePathStyle` and `s3Url` defined, a region does not need to be defined inside BSL.

### What happened?

This velero CR should not have error status
```yaml
apiVersion: oadp.openshift.io/v1alpha1
kind: Velero
metadata:
  name: example-velero
  namespace: openshift-adp
spec:
  backupStorageLocations:
    - backupSyncPeriod: 2m0s
      config:
        profile: default
        s3ForcePathStyle: 'true'
        s3Url: 'http://s3.openshift-storage.svc/'
      credential:
        key: cloud
        name: cloud-credentials
      default: true
      objectStorage:
        bucket: migstorage
        prefix: velero
      provider: aws
  defaultVeleroPlugins:
    - openshift
    - aws
    - csi
  enableRestic: true
  veleroFeatureFlags:
    - EnableCSI
  volumeSnapshotLocations:
    - config:
        profile: default
        region: us-east-1
      provider: aws
status:
  conditions:
    - lastTransitionTime: '2021-10-29T18:03:27Z'
      message: >-
        prefix and region for AWS backupstoragelocation object storage cannot be
        empty. It is required for backing up images
      reason: Error
      status: 'False'
      type: Reconciled

```

### OADP Version

0.4.x (Beta)

### OpenShift Version

4.8

### New issue

- [X] This issue is new

**Comment by kaovilai** [comment]
For context, this CR is for oadp-lab which always use local object storage but in one of the exercises uses EBS CSI snapshot. (it may not even need the VSL defined).

**Comment by kaovilai** [comment]
Solution:
1. Remove BSL region requirement
2. If s3url is not defined, query region (without requirring user input) for. registry if backupImages=true
3. if s3url  is defined and backupImages=true, provide a default (does not have to be valid aws region, not checked)region  envvar string to the docker registry s3 driver

Implement like [upstream](https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/60eb709839c0f70fb38d8c8eb4573a28834e32fa/velero-plugin-for-aws/object_store.go#L134-L141) using GetBucketRegion [function](https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/60eb709839c0f70fb38d8c8eb4573a28834e32fa/velero-plugin-for-aws/helpers.go#L31)

**Comment by kaovilai** [comment]
Title updated based on info from
https://github.com/vmware-tanzu/velero/issues/4305

**Comment by kaovilai** [comment]
Waiting on behavior confirmation when 
- https://github.com/vmware-tanzu/velero-plugin-for-aws/pull/91
is closed or merged
